### ISSUE TITLE
'Select all' checkbox is missing the visual focus outline when the state is indeterminate (half-checked)

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri-design-system#974]
## Product
[Originally experienced in Kolibri](https://github.com/learningequality/kolibri/issues/13145), will probably be present in Studio where KCheckbox is used.

## Expected behavior
Checkbox element must display the visual outline whenever the user who navigates by keyboard encounters it and it receives the focus by tabbing into it. State of the element (checked, unchecked or indeterminate) should not have effect on the elements outline when in focus.

## Actual behavior
https://github.com/user-attachments/assets/8032b5e8-41e8-4cb7-82a9-f68d4ec7e223

Outline to indicate that the `Select all` checkbox is in focus appears correctly for both the unchecked and fully checked states, and is missing ONLY if the state is half-partially checked. So to replicate it you will need to activate the checkbox for at least one of the resources and navigate back to the `Select all` checkbox. It is functional, and correctly announced by the screen reader, it just misses the outline (which is a barrier for sighted users who navigate by keyboard, if not for those who use the screen reader 🙂)

## Steps to reproduce the issue
1. In Kolibri, navigate to any page that presents a list of selectable resources (in **Coach > Lesson/Quiz > Manage resources**, or **Device > Channels > Import**).
2. Activate at least one checkbox (select to add a resource).
3. **Navigate back by keyboard** using the `Shift+Tab` and observe that you do not see the outline when the  `Select all` checkbox is in focus.

## Environment
- OS: all
- Browser version: all

**Comment by alokdevx** [comment]
I would like to work on this issue @radinamatic

**Comment by alokdangre** [comment]
I would like to work on these issue @radinamatic  
So plzz can you assign me this issue 
I have a good understanding of it 
I will complete it as soon as possible

**Comment by MisRob** [comment]
Hi @alokdangre, welcome and thanks for volunteering! We can only assign 'help wanted' issues and there are no such free issues right now. Also please have a look at CONTRIBUTING.md if you haven't yet done so.

**Comment by AlexVelezLl** [comment]
Closed by #983
